crates/bevy_reflect/src/list.rs

```diff
@@ -401,7 +401,7 @@ impl<'a> Iterator for ListIter<'a> {
     #[inline]
     fn next(&mut self) -> Option<Self::Item> {
         let value = self.list.get(self.index);
-        self.index += 1;
+        self.index += value.is_some() as usize;
```

Comment:
Is this the right behavior? I would expect this to return `None` during an attempted overflow, not return the last item forever.

Reply:
if current value is something go to next, right?

Reply:
> Is this the right behavior? I would expect this to return `None` during an attempted overflow, not return the last item forever.

I don't see how this code would return the last element forever.

If we call `get` on the last element, `index` will be incremented one past the last element. The next time `get` is called it will be passed a value one past the last element and it should return a None.